Mathlib/Topology/Category/LightProfinite/IsLight.lean

```diff
@@ -0,0 +1,266 @@
+/-
+Copyright (c) 2023 Dagur Asgeirsson. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Dagur Asgeirsson
+-/
+import Mathlib.CategoryTheory.Limits.Shapes.Countable
+import Mathlib.Topology.Category.LightProfinite.Basic
+import Mathlib.Topology.Category.Profinite.AsLimit
+import Mathlib.Topology.Category.Profinite.CofilteredLimit
+import Mathlib.Topology.ClopenBox
+/-!
+# Being light is a property of profinite spaces
+
+We define a class `Profinite.IsLight` which says that a profinite space `S` has countably many
+clopen subsets. We prove that a profinite space which `IsLight` gives rise to a `LightProfinite` and
+that the underlying profinite space of a `LightProfinite` is light.
+
+## Main results
+
+* `IsLight` is preserved under taking cartesian products
+
+* Given a monomorphism whose target `IsLight`, its sourse is also light: `Profinite.isLight_of_mono`
+
+## Project
+
+* Prove the Stone duality theorem that `Profinite` is equivalent to the opposite category of
+  boolean algebras. Then the property of being light says precisely that the corresponding
+  boolean algebra is countable. Maybe constructions of limits and colimits in `LightProfinite`
+  becomes easier when transporting over this equivalence.
+
+-/
+
+universe u
+
+open CategoryTheory Limits FintypeCat Opposite TopologicalSpace
+
+open scoped Classical
+
+/-- A profinite space *is light* if it has countably many clopen subsets.  -/
+class Profinite.IsLight (S : Profinite) : Prop where
+  /-- The set of clopens is countable -/
+  countable_clopens : Countable (Clopens S)
+
+attribute [instance] Profinite.IsLight.countable_clopens
+
+instance (X Y : Profinite) [X.IsLight] [Y.IsLight] : (Profinite.of (X × Y)).IsLight where
+  countable_clopens := Clopens.countable_prod
+
+instance (S : Profinite) [S.IsLight] : Countable (DiscreteQuotient S) := by
```

Comment:
Can you add a mathematical explanation of this proof? (In principle it would be nice to split it...)

Reply:
I added just a small explanation, but I agree it would be nice to split it. I'll see what I can do about that tomorrow.

Reply:
In order to split the proof, I would suggest constructing, for any `S` profinite, an equiv. between `DiscreteQuotient S` and suitable `Finset` of clopens.

Reply:
I changed the proof to using an injective map `DiscreteQuotient S → Finset (Clopens S)` instead, I think it's a lot clearer. I added a TODO to prove that this gives a bijection with the finite partitions of `S` into clopens.